Sources/XcbeautifyLib/JunitReporter.swift

```diff
@@ -158,6 +162,31 @@ private enum JunitComponent {
     case skippedTest(TestCase)
 }
 
+private extension JunitComponent {
+    var testCase: TestCase? {
+        switch self {
+        case .testSuiteStart: nil
+        case let .failingTest(testCase), let .testCasePassed(testCase), let .skippedTest(testCase):
+            testCase
+        }
+    }
+}
+
+private extension [JunitComponent] {
+    mutating func removePreviousFailingTestsAfter(_ testCase: TestCase) {
+        // base case, empty array or last is not the last passed test case
+        guard let previousTestCase = last?.testCase,
+              testCase.classname == previousTestCase.classname,
+              testCase.name == previousTestCase.name
+        else {
+            return
+        }
```

Comment:
The method name says `removePreviousFailing`, but the `guard` doesn't actually consider if the last test is a failing one. It'd need to check that `testCase.failure` is non-`nil`.

Reply:
Indeed, this is required for `Run tests repeatedly`